pkg/roachprod/vm/ibm/helpers.go

```diff
@@ -0,0 +1,1205 @@
+		}
+	}
+
+	return nil
+
+}
+
+// checkAttachedVolumeTypeSizeIops checks if the given volume type,
+// and IOPS combination is valid based on a static map of configurations.
+// It returns an error if the combination is not valid.
+func (p *Provider) checkAttachedVolumeTypeSizeIops(t string, s int, i int) error {
+	type profileCapacities struct {
+		sizeRange           string
+		iopsRange           func(i int) string
+		customIopsSupported bool
+	}
+	availableTypes := map[string]profileCapacities{
+		"general-purpose": {
+			sizeRange: "[10-160000]",
+			iopsRange: func(_ int) string { return "[3000-48000]" },
+		},
+		"5iops-tier": {
+			sizeRange: "[10-9600]",
+			iopsRange: func(_ int) string { return "[3000-48000]" },
+		},
+		"10iops-tier": {
+			sizeRange: "[10-4800]",
+			iopsRange: func(_ int) string { return "[3000-48000]" },
+		},
+		"custom": {
+			sizeRange: "[10-160000]",
+			iopsRange: func(s int) string {
+				switch {
+				case s < 40:
+					return "[100-1000]"
+				case s < 80:
+					return "[100-2000]"
+				case s < 100:
+					return "[100-4000]"
+				case s < 500:
+					return "[100-6000]"
+				case s < 1000:
+					return "[100-10000]"
+				case s < 2000:
+					return "[100-20000]"
+				case s < 4000:
+					return "[200-40000]"
+				case s < 8000:
+					return "[300-40000]"
+				case s < 10000:
+					return "[500-48000]"
+				default:
+					return "[1000-48000]"
+				}
+			},
+			customIopsSupported: true,
+		},
+	}
+
+	if _, ok := availableTypes[t]; !ok {
+		return fmt.Errorf("volume type %s is not supported", t)
+	}
+
+	// Check if the size is in range
+	sizeRange, err := p.valInRange(s, availableTypes[t].sizeRange)
+	if err != nil {
+		return errors.Wrap(err, "unable to check if volume size is in range")
+	}
+	if !sizeRange {
+		return fmt.Errorf("volume size %d is out of range %s", s, availableTypes[t].sizeRange)
+	}
+
+	// Check if the IOPS is in range
+	if i > 0 {
+		if !availableTypes[t].customIopsSupported {
+			return fmt.Errorf("custom IOPS is not supported for volume type %s", t)
+		}
+
+		iopsRangeForTierSize := availableTypes[t].iopsRange(s)
+		iopsRange, err := p.valInRange(i, iopsRangeForTierSize)
+		if err != nil {
+			return errors.Wrap(err, "unable to check if volume IOPS is in range")
+		}
+		if !iopsRange {
+			return fmt.Errorf("volume IOPS %d is out of range %s", i, iopsRangeForTierSize)
+		}
+	}
+	return nil
+}
+
+// valInRange checks if the given integer value is in the range specified by the
+// range string. Range string should be in the format "[lower,upper)".
+// Lower and upper bounds can be empty, and both inclusive and exclusive bounds
+// are supported.
+func (p *Provider) valInRange(value int, rangeStr string) (bool, error) {
+	if len(rangeStr) < 3 {
+		return false, fmt.Errorf("range too short")
+	}
+	lowerInclusive, upperInclusive := rangeStr[0] == '[', rangeStr[len(rangeStr)-1] == ']'
+	parts := strings.Split(rangeStr[1:len(rangeStr)-1], "-")
```

Comment:
If `rangeStr` is of the form `24-56`, the parses interprets it as `lower = 4` and `upper = 5`.